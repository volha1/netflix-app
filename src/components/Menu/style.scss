@import '../../common/scss/base.scss';

.menu {
  width: 200px;
  padding: 20px 0px;
  position: absolute;
  top: 10px;
  right: 10px;
  background-color: $black;
  display: none;

  &.active {
    display: block;
  }

  button {
    width: 100%;
    background-color: $black;
    border-radius: 0;
    padding: 15px;

    &:hover {
      background-color: $red;
    }
  }

  .close-btn::after {
    font-size: 34px;
  }
}
