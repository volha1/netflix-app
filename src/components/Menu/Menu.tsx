import React, { ReactElement, useContext, useCallback } from 'react';
import { useDispatch } from 'react-redux';
import classNames from 'classnames';
import Context from '../../context/Context';
import CloseBtn from '../CloseBtn';
import { markMovieForDeletion, saveMovieForEditing } from '../../store/moviesSlice';
import './style.scss';
import Movie from '../../entity/Movie';

type SortingProps = {
  isVisible: boolean;
  onClose: () => void;
  movie: Movie;
};

const Menu = ({ isVisible, onClose, movie }: SortingProps): ReactElement => {
  const classes = classNames('menu', { active: isVisible });
  const [handleEditMovieForm, handleDeleteMovieMessage] = useContext(Context);
  const dispatch = useDispatch();

  const handleEditBtn = useCallback((): void => {
    dispatch(saveMovieForEditing(movie));
    onClose();
    handleEditMovieForm();
  }, []);

  const handleDeleteBtn = useCallback((): void => {
    dispatch(markMovieForDeletion(movie.id));
    onClose();
    handleDeleteMovieMessage();
  }, []);

  return (
    <div
      className={classes}
      onClick={(event: React.MouseEvent<HTMLDivElement, MouseEvent>): void => {
        event.stopPropagation();
      }}
    >
      <CloseBtn onClose={onClose} />
      <ul>
        <li>
          <button type="button" onClick={handleEditBtn}>
            Edit
          </button>
        </li>
        <li>
          <button type="button" onClick={handleDeleteBtn}>
            Delete
          </button>
        </li>
      </ul>
    </div>
  );
};

export default Menu;
