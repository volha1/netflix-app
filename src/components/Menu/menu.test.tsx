import React from 'react';
import renderer from 'react-test-renderer';

import Menu from './Menu';

it('Menu snapshot', () => {
  const onClose = jest.fn();
  const onDelete = jest.fn();
  const onEdit = jest.fn();
  const component = renderer.create(<Menu isVisible onClose={onClose} onDelete={onDelete} onEdit={onEdit} />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
