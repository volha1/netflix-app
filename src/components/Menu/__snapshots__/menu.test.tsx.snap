// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu snapshot 1`] = `
<div
  className="menu active"
  onClick={[Function]}
>
  <div
    className="close-btn"
    onClick={[MockFunction]}
  />
  <ul>
    <li>
      <button
        onClick={[MockFunction]}
        type="button"
      >
        Edit
      </button>
    </li>
    <li>
      <button
        onClick={[MockFunction]}
        type="button"
      >
        Delete
      </button>
    </li>
  </ul>
</div>
`;
